packages/frontend/src/hooks/useUsdValue.ts

```diff
@@ -0,0 +1,31 @@
+import Big from 'big.js';
+import { useEffect, useState } from 'react';
+import { useSifi } from 'src/providers/SDKProvider';
+
+type UseUsdPriceOptions = {
+  address?: string | null;
+  chainId?: number | null;
+  amount?: string;
+};
+
+const useUsdValue = ({ address, chainId, amount }: UseUsdPriceOptions) => {
+  const [usdPrice, setUsdPrice] = useState<string>();
+  const sifi = useSifi();
+
+  const fetchUsdPrice = async () => {
+    if (!address || !chainId) return;
+
+    const response = await sifi.getUsdPrice(chainId, address);
```

Comment:
Missing explicit error handling on this call.

We could also just consider refactoring this hook to use react-query?